Simplify todo list feature.

troessner opened the pull request:
Implements the first part of #1411

troessner commented:
Ready for review!

mvz commented on lib/reek/cli/command/todo_list_command.rb:
```diff
@@ -12,17 +12,19 @@ module Command
       class TodoListCommand < BaseCommand
-        FILE_NAME = '.todo.reek'
+        DEFAULT_CONFIGURATION_FILE_NAME = Configuration::ConfigurationFileFinder::DEFAULT_FILE_NAME
+        DETECTOR_KEY = Configuration::AppConfiguration::DETECTORS_KEY
```
This level of indirection seems undesirable to me. I have a feeling that if it seems necessary, it probably makes sense for these constants to live in some module where they're readily accessible for all modules.

mvz commented on lib/reek/cli/command/todo_list_command.rb:
```diff
@@ -44,6 +46,14 @@ def groups
+            configuration_file.write HEADER
+            configuration_file.write({ DETECTOR_KEY => groups }.to_yaml)
+          end
+          puts "\n'#{DEFAULT_CONFIGURATION_FILE_NAME}' generated! You can now use this as a starting point."
```
I'd move the puts out of `#write_to_file`, so all output is concentrated in `#execute`.

mvz commented on lib/reek/cli/command/todo_list_command.rb:
```diff
@@ -12,17 +12,19 @@ module Command
           if smells.empty?
-            puts "\n'.todo.reek' not generated because "\
-                    'there were no smells found!'
+            puts DO_NOTHING_MESSAGE
```
Let's rename this to `NO_SMELLS_FOUND_MESSAGE` to match `EXISTING_FILE_MESSAGE`, since both messages indicate a situation where nothing is done.

troessner commented:
@mvz I addressed all of your comments and I moved all other key names out of AppConfiguration into Reek::Constants. As you can see in the diff, all of them were used outside of AppConfiguration already  so this feels like a good idea even regardless of this PR.
Please re-review.

mvz commented on lib/reek/constants.rb:
```diff
@@ -0,0 +1,13 @@
+
+module Reek
+  #
+  # This module holds some constants that are used across Reek.
```
If we move these up into `Reek` we don't have to namespace them in the rest of the code. WDYT?

troessner replied:
Yes, great idea, will change that.

mvz commented on spec/reek/cli/command/todo_list_command_spec.rb:
```diff
@@ -2,67 +2,95 @@
+      context 'when exists already' do
+        it 'does not update the configuration' do
+          write_existing_configuration default_configuration_file_name
+          command = todo_command configuration: default_configuration
```
Does it matter for the test what is passed in `configuration`?

troessner replied:
Good catch, removed.

mvz commented on spec/reek/cli/command/todo_list_command_spec.rb:
```diff
@@ -2,67 +2,95 @@
+    end
+
+    def write_existing_configuration(file_name)
+      File.write(file_name, existing_configuration)
```
I'm not convinced this needs to be a method, since it is only called once. How about inlining it?

troessner replied:
Ack

mvz commented on spec/reek/cli/command/todo_list_command_spec.rb:
```diff
@@ -2,67 +2,95 @@
-        expect(File).not_to have_received(:write)
-      end
+    def write_smelly_file
+      File.write(SMELLY_FILE.basename, smelly_file)
```
I'm not convinced this needs to be a method, since it is only called once. How about inlining it?

troessner replied:
Ack

troessner commented:
Addressed all your comments @mvz ;)

troessner commented:
@mvz I'll be in lovely Israel for the next 2 weeks without a laptop starting tomorrow, would be great if you could re-review today so we can hopefully merge this one before ;)

mvz commented on lib/reek/constants.rb:
```diff
@@ -0,0 +1,13 @@
+  #
+  # This module holds some constants that are used across Reek.
+  #
+  module Constants
```
This module can go now, right?

mvz commented:
Just one more thing and then it's good to go.

troessner commented:
Fixed ;)

troessner commented on .simplecov:
```diff
@@ -9,6 +9,6 @@ end
 SimpleCov.at_exit do
   SimpleCov.result.format!
-  SimpleCov.minimum_coverage 98.9
+  SimpleCov.minimum_coverage 98.88
```
I feel like this PR is well tested enough :)
